pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractBaseDispatcher.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pulsar.broker.service;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.util.concurrent.FastThreadLocal;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.bookkeeper.mledger.Entry;
+import org.apache.bookkeeper.mledger.impl.PositionImpl;
+import org.apache.pulsar.common.api.Commands;
+import org.apache.pulsar.common.api.proto.PulsarApi.CommandAck.AckType;
+import org.apache.pulsar.common.api.proto.PulsarApi.MessageMetadata;
+
+public abstract class AbstractBaseDispatcher {
+
+    protected final Subscription subscription;
+
+    protected AbstractBaseDispatcher(Subscription subscription) {
+        this.subscription = subscription;
+    }
+
+    /**
+     * Filter messages that are being sent to a consumers.
+     * <p>
+     * Messages can be filtered out for multiple reasons:
+     * <ul>
+     * <li>Checksum or metadata corrupted
+     * <li>Message is an internal marker
+     * <li>Message is not meant to be delivered immediately
+     * </ul>
+     *
+     * @param entries
+     *            a list of entries as read from storage
+     *
+     * @param batchSizes
+     *            an array where the batch size for each entry (the number of messages within an entry) is stored. This
+     *            array needs to be of at least the same size as the entries list
+     *
+     * @param sendMessageInfo
+     *            an object where the total size in messages and bytes will be returned back to the caller
+     * @param subscription
+     *            the subscription object
+     */
+    public void filterEntriesForConsumer(List<Entry> entries, int[] batchSizes, SendMessageInfo sendMessageInfo) {
```

Comment:
on what bases are we going to filter the entry? do we have plan to pass Consumer to apply function for filtering?

Reply:
Current plan is to have a general way to filter out messages for (corrupted, delayed messages, internal markers). 

It could be also used, in future, to do some other filtering as well, though that's not the intention here to generalize for that.